File: backend/app/services/hearing_store.py
from app.clients.mongo import get_db
from app.models.hearing import HearingState


def create(state: HearingState) -> None:
	get_db()["hearings"].replace_one(
		{"hearing_id": state.hearing_id},
		state.model_dump(),
		upsert=True,
	)


def get(hearing_id: str) -> HearingState | None:
	doc = get_db()["hearings"].find_one({"hearing_id": hearing_id}, {"_id": 0})
	if doc is None:
		return None
	return HearingState(**doc)


def update(state: HearingState) -> None:
	create(state)

File: frontend/lib/services/hearingStore.ts
import { HearingState } from '@/types/hearing';
import { getDb } from '@/lib/mongo';

export const createHearing = async (state: HearingState): Promise<void> =>
{
	const db = await getDb();
	await db.collection('hearings').replaceOne(
		{ hearing_id: state.hearing_id },
		state,
		{ upsert: true }
	);
};

export const getHearing = async (hearingId: string): Promise<HearingState | null> =>
{
	const db = await getDb();
	const doc = await db.collection('hearings').findOne(
		{ hearing_id: hearingId },
		{ projection: { _id: 0 } }
	);
	if(doc === null) return null;
	return doc as unknown as HearingState;
};

export const updateHearing = async (state: HearingState): Promise<void> =>
{
	await createHearing(state);
};
